src/charm.py

```diff
@@ -0,0 +1,595 @@
+                logger.error("Invalid path %s", path)
+                return None
+
+            owner, repo = paths
+            path = GitHubRepo(owner=owner, repo=repo)
+        else:
+            path = GitHubOrg(org=path, group=self.config["group"])
+
+        app_name, unit = self.unit.name.rsplit("/", 1)
+        return RunnerManager(
+            app_name,
+            unit,
+            RunnerManagerConfig(path, token, "jammy", self.service_token),
+            proxies=self.proxies,
+        )
+
+    @catch_unexpected_charm_errors
+    def _on_install(self, _event: InstallEvent) -> None:
+        """Handle the installation of charm.
+
+        Args:
+            event: Event of installing charm.
+        """
+        self.unit.status = MaintenanceStatus("Installing packages")
+
+        try:
+            # The `_start_services`, `_install_deps` includes retry.
+            self._install_deps()
+            self._start_services()
+        except SubprocessError as err:
+            logger.exception(err)
+            # The charm cannot proceed without dependencies.
+            self.unit.status = BlockedStatus("Failed to install dependencies")
+            return
+
+        runner_manager = self._get_runner_manager()
+        if runner_manager:
+            self.unit.status = MaintenanceStatus("Downloading runner binary")
+            try:
+                runner_info = runner_manager.get_latest_runner_bin_url()
+                logger.info(
+                    "Downloading %s from: %s", runner_info.filename, runner_info.download_url
+                )
+                self._stored.runner_bin_url = runner_info.download_url
+                runner_manager.update_runner_bin(runner_info)
+            # Safe guard against transient unexpected error.
+            except Exception as err:  # pylint: disable=broad-exception-caught
+                logger.exception("Failed to update runner binary")
+                # Failure to download runner binary is a transient error.
+                # The charm automatically update runner binary on a schedule.
+                self.unit.status = MaintenanceStatus(f"Failed to update runner binary: {err}")
+                return
+            self.unit.status = MaintenanceStatus("Starting runners")
+            try:
+                self._reconcile_runners(runner_manager)
+                self.unit.status = ActiveStatus()
+            except RunnerError as err:
+                logger.exception("Failed to start runners")
+                self.unit.status = MaintenanceStatus(f"Failed to start runners: {err}")
+        else:
+            self.unit.status = BlockedStatus("Missing token or org/repo path config")
+
+    @catch_unexpected_charm_errors
+    def _on_upgrade_charm(self, _event: UpgradeCharmEvent) -> None:
+        """Handle the update of charm.
+
+        Args:
+            event: Event of charm upgrade.
+        """
+        logger.info("Reinstalling dependencies...")
+        self._install_deps()
```

Comment:
Shouldn't the SubprocessError be handled here like it is on _on_install?

Reply:
Yes, the error handling is missing.